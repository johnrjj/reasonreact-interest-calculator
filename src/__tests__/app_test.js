// Generated by BUCKLESCRIPT VERSION 1.8.0, PLEASE EDIT WITH CARE
'use strict';

var App                  = require("../app.js");
var Jest                 = require("bs-jest/lib/js/src/jest.js");
var Curry                = require("bs-platform/lib/js/curry.js");
var ReasonReact          = require("reason-react/lib/js/src/reasonReact.js");
var ReactShallowRenderer = require("bs-react-test-renderer/lib/js/src/reactShallowRenderer.js");

describe("src/app.re", (function () {
        Jest.test("is defined", (function () {
                var component = ReactShallowRenderer.renderWithRenderer(ReasonReact.element(/* None */0, /* None */0, App.make(/* array */[])));
                return Curry._1(Jest.ExpectJs[/* toBeDefined */23], Curry._1(Jest.ExpectJs[/* expect */0], component));
              }));
        return Jest.test("renders", (function () {
                      var tree = ReactShallowRenderer.renderWithRenderer(ReasonReact.element(/* None */0, /* None */0, App.make(/* array */[])));
                      return Curry._1(Jest.ExpectJs[/* toMatchSnapshot */15], Curry._1(Jest.ExpectJs[/* expect */0], tree));
                    }));
      }));

/*  Not a pure module */
