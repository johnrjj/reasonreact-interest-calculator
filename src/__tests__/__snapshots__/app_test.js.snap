// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/app.re renders 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <h1>
      ReasonReact Interest Calculator
    </h1>
  </div>
  <label>
    Principle
    <input
      onChange={[Function]}
      value="1000"
    />
  </label>
  <label>
    Rate
    <input
      onChange={[Function]}
      value="0.03"
    />
  </label>
  <label>
    Years to grow
    <input
      onChange={[Function]}
      value="10"
    />
  </label>
  <label>
    Total
    <input
      readOnly={true}
      value="1349.35354719"
    />
  </label>
</div>
`;
